### PR TITLE
Add abstraction to convert between `xdr.SignerKey`s and signers (strkeys)

### PR DESCRIPTION
**Pull request by Shaptic**

### CODE REVIEW

File: test/unit/signerkey_test.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
+describe('SignerKey', function() {
+  describe('encode/decode roundtrip', function() {
+    const TEST_CASES = [
+      {
```
Comment (paulbellamy) [inline test/unit/signerkey_test.js:4]: `s` and `m` test cases for completeness?

File: src/signerkey.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
+import xdr from './generated/stellar-xdr_generated';
+import { StrKey, encodeCheck, decodeCheck } from './strkey';
+
+export class SignerKey {
+  static decodeAddress(address) {
```
Comment (paulbellamy) [inline src/signerkey.js:5]: js comments with description and types?

File: src/signerkey.js [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+      case 'ed25519PublicKey': // falls through
+      case 'preAuthTx': // falls through
+      case 'sha256Hash': // falls through
+      default:
+        return encoder(raw);
+    }
+  }
+
+  static encodeSignerKey(signerKey) {
```
Comment (paulbellamy) [inline src/signerkey.js:37]: comment with description and types here as well?

File: src/signerkey.js [hunk 4]
```diff
@@ -0,0 +1,68 @@
+import xdr from './generated/stellar-xdr_generated';
+import { StrKey, encodeCheck, decodeCheck } from './strkey';
+
+export class SignerKey {
```
Comment (paulbellamy) [inline src/signerkey.js:4]: I guess this doesn't need types as it's not meant for direct usage?

**Review by paulbellamy — APPROVED** [review]
A few nits and suggestions, but lgtm.

File: src/signerkey.js (hunk 2 above)
Comment (Shaptic) [reply src/signerkey.js:23]: https://github.com/stellar/js-stellar-base/pull/520/commits/342a98758cee7584217107db02ece7624a9d4f54

File: src/signerkey.js (hunk 4 above)
Comment (Shaptic) [reply src/signerkey.js:13]: True, but done anyway in https://github.com/stellar/js-stellar-base/pull/520/commits/342a98758cee7584217107db02ece7624a9d4f54

File: src/signerkey.js (hunk 3 above)
Comment (Shaptic) [reply src/signerkey.js:61]: Yep https://github.com/stellar/js-stellar-base/pull/520/commits/342a98758cee7584217107db02ece7624a9d4f54

**Pull request merged**